isis/src/base/objs/OriginalLabel/OriginalLabel.cpp

```diff
@@ -16,19 +16,40 @@ namespace Isis {
   /**
    * Constructor for creating an original blob with a given name
    */
-  OriginalLabel::OriginalLabel() : Isis::Blob("IsisCube", "OriginalLabel") {
+  OriginalLabel::OriginalLabel() {
     m_originalLabel.setTerminator("");
   }
 
+
+  /**
+   * Constructor for creating an original label from a blob object.
+   */
+  OriginalLabel::OriginalLabel(Isis::Blob &blob) {
+    stringstream os;
+    char* dataBuffer = blob.getBuffer();
+    for(int i = 0; i < blob.Size(); i++) {
+        os << dataBuffer[i];
+    }
+    os >> m_originalLabel;
+  }
+
   /**
    * Constructor for creating an original blob with a given name and file to
    * read labels from.
    *
    * @param file File to read labels from
    */
-  OriginalLabel::OriginalLabel(const QString &file) :
-    Isis::Blob("IsisCube", "OriginalLabel") {
-    Blob::Read(file);
+  OriginalLabel::OriginalLabel(const QString &file){
+    Pvl pvl;
+    stringstream os;
+    Blob blob = Blob("IsisCube", "OriginalLabel");
+    blob.Read(file);
+    char *buff = blob.getBuffer();
```

Comment:
From here down is the same logic as the Blob constructor. We should move this logic into a shared function, it could be public or private.

Reply:
Yes but this may not always be the case and I don't know where a function like that would live. If anything this may be reason to have history and original label exist as the same object